### PR TITLE
Fix uv_os_tmpdir when env variable is 260 chars

### PR DESCRIPTION
**Pull request by musm**
When TMP environment variable is 261 chars (including the trailing slash) or 260 chars without the trailing slash, `uv_os_tmpdir` throws an `UV_EIO` error which it shouldn't, since the total input string size allowable for this function is PATH_MAX+1 (including the trailing slash).

Please see the  Julia code in https://github.com/libuv/libuv/issues/2336 which demonstrates the problem. 

In any case, there is also no harm in setting the buffer larger.

### CODE REVIEW

File: src/win/util.c [hunk 1]
```diff
@@ -1171,18 +1171,18 @@ int uv_os_homedir(char* buffer, size_t* size) {
...
-  wchar_t path[MAX_PATH + 1];
+  wchar_t path[MAX_PATH + 2];
   DWORD bufsize;
   size_t len;
 
   if (buffer == NULL || size == NULL || *size == 0)
     return UV_EINVAL;
 
-  len = GetTempPathW(MAX_PATH + 1, path);
+  len = GetTempPathW(MAX_PATH + 2, path);
```
Comment (cjihrig) [inline src/win/util.c]: Unrelated to the issue at hand, but maybe this and line 1185 could use `sizeof`.

**Comment by musm** [comment]
Good to merge?

**Comment by cjihrig** [comment]
I'd like some collaborators more familiar with Windows to sign off on this. It seems like an odd edge case that I can't really test out for myself.

File: src/win/util.c (hunk 1 above)
Comment (bnoordhuis) [reply src/win/util.c]: It should be `ARRAY_SIZE(path)` because it's an array of `wchar_t` but otherwise I agree: DRY.

**Comment by musm** [comment]
Exactly, the MSDN documentation is wrong and has different behavior as demonstrated in my test scripts, which show that it can actually report back a buffer length of 262 and not 261 as stated in the docs, when the `TMP` environment variable is 261 chars long (including the trailing backslash).

**Comment by bzoz** [comment]
@musm, can you create a test for this? Otherwise this LGTM.

**Comment by musm** [comment]
@bzoz I'm not sure how to test thing. Would anyone else be able to create a test case based on the linked issue?

I confirmed that building this branch and the corresponding  julia code is fixed.

Here's how the test should look like.
- Set the environment variable TMP to a string on length 261: e.g. "C:\\xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\\"  (must have the C:\\ prefix and  trailing backslash)
- Ensure this function does not throw  `UV_EIO`
- Assert the length returned in equal to the length of the TMP var.

**Comment by musm** [comment]
Test added.  They pass on this branch but on throw the branch libuv:v1.x throw an assertion.

**Comment by musm** [comment]
closes https://github.com/libuv/libuv/issues/2336

**Comment by bzoz** [comment]
CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/1446/

**Comment by musm** [comment]
Thanks, has CI passed? I can't seem to understand the linked CI page.

**Comment by bzoz** [comment]
3 test have failed on win 2008r2:
```
05:50:23  not ok 100 - fs_symlink
05:50:23  # exit code 3
05:50:23  # Output from process `fs_symlink`:
05:50:23  # Assertion failed in test-fs.c on line 1944: stricmp(req.ptr, test_file_abs_buf) == 0
05:50:23  not ok 101 - fs_symlink_dir
05:50:23  # exit code 3
05:50:23  # Output from process `fs_symlink_dir`:
05:50:23  # r == 0
05:50:23  # Assertion failed in test-fs.c on line 2106: strlen(req.ptr) == test_dir_abs_size - 5
05:50:23  not ok 102 - fs_symlink_junction
05:50:23  # exit code 3
05:50:23  # Output from process `fs_symlink_junction`:
05:50:23  # r == 0
05:50:23  # Assertion failed in test-fs.c on line 2106: strlen(req.ptr) == test_dir_abs_size - 5
```
Those look unrelated though, I'll land this PR later today or tomorrow morning.

**Comment by cjihrig** [comment]
@bzoz those failures are https://github.com/libuv/libuv/issues/2303, so almost certainly unrelated.

**Comment by musm** [comment]
Great, thanks all. Glad to have my first PR to libuv land!

**Comment by bzoz** [comment]
Landed in https://github.com/libuv/libuv/commit/a5d374377822ddaa1961bddcef4bd53189c83d88. Great job @musm!
